src/legacy/core_plugins/kibana/plugin_dependencies.d.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * The idea here is to provide a temporary home for defining dependency
+ * types that are passed to plugins in the new platform.
+ *
+ * When a plugin is being shimmed and requires a dependency not listed here,
+ * it should be added to the list so the same name is used consistently in
+ * future shims. This will also help us to identify what the public contracts
+ * should be in our new platform plugins.
+ *
+ * Once shims are fully removed and everything is migrated, this file can be
+ * deleted and the relevant types can be directly imported from each plugin.
+ */
+export interface Ui {
+  readonly createLegacyClass: any;
+  readonly savedObjectLoader: any;
+  readonly savedObjectProvider: any;
+  readonly savedObjectsClientProvider: any;
+  readonly uiModules: any;
+}
+
+export interface Management {
+  readonly savedObjectManagementRegistry: any;
+}
+
+export interface Visualize {
+  readonly updateOldState: any;
+  readonly visProvider: any;
+  readonly visTypesRegistryProvider: any;
+}
+
+export interface Dependencies {
+  // ui/public stuff
+  readonly ui?: Ui;
+
+  // new platform plugins (oss)
+  readonly dashboard?: {};
+  readonly dataAccess?: {};
+  readonly discover?: {};
+  readonly embeddables?: {};
+  readonly globalSearchSource?: {};
+  readonly home?: {};
+  readonly interpreter?: {};
+  readonly management?: Management;
+  readonly visEditor?: {};
+  readonly visualize?: Visualize;
+
+  // new platform plugins (x-pack)
+  readonly dashboardMode?: {};
+  readonly graph?: {};
+  readonly kueryAutocomplete?: {};
+  readonly reporting?: {};
+  readonly telemetry?: {};
+}
```

Comment:
Where should this live, if we are wanting to share it across all plugins? Should it be exported under the `Legacy` namespace from `kibana.d.ts` for now?

Reply:
I would export this from `core_plugins/kibana/public/index.d.ts` and then plugins would import it via relative import from their own public code.

Reply:
Think of it this way: these are part of the public interface of the kibana plugin's client-side code. So they should be exported via the root of that module, which is core_plugins/kibana/public.

Reply:
Not sure if we are thinking about the same thing here. Why is this part of the Kibana plugin's interface? This is the typing for what the Kibana plugin takes into as dependencies?

Reply:
Yeah, the original purpose for maintaining typings for dependencies was to ensure consistency of the structure of the `dependencies` object as we shim plugins... that way if multiple people are working on this, we don't need to worry about the structure of that object diverging across shims. (In the real new platform, this isn't really an issue as core handles grabbing all of your dependencies for you -- it's only a concern when shimming as there's nothing stopping you from structuring these however you want).

The intent was for this to be temporary to assist in the shimming process, as these should be replaced with real types that are exported from each plugin (once those exist of course).

I think we can revisit this a bit later; if we shift focus to cleaning up `ui/public` and sorting out what our new platform contracts should be, this step may become less important.